plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelTracingTelemetry.java

```diff
@@ -46,23 +50,55 @@ public void close() {
     }
 
     @Override
-    public Span createSpan(String spanName, Span parentSpan) {
-        return createOtelSpan(spanName, parentSpan);
+    public Span createSpan(String spanName, Span parentSpan, Attributes attributes) {
+        return createOtelSpan(spanName, parentSpan, attributes);
     }
 
     @Override
     public TracingContextPropagator getContextPropagator() {
         return new OTelTracingContextPropagator(openTelemetry);
     }
 
-    private Span createOtelSpan(String spanName, Span parentSpan) {
-        io.opentelemetry.api.trace.Span otelSpan = otelSpan(spanName, parentSpan);
+    private Span createOtelSpan(String spanName, Span parentSpan, Attributes attributes) {
+        io.opentelemetry.api.trace.Span otelSpan = otelSpan(spanName, parentSpan, convertToAttributes(attributes));
         return new OTelSpan(spanName, otelSpan, parentSpan);
     }
 
-    io.opentelemetry.api.trace.Span otelSpan(String spanName, Span parentOTelSpan) {
+    private io.opentelemetry.api.common.Attributes convertToAttributes(Attributes attributes) {
+        if (attributes != null) {
+            AttributesBuilder attributesBuilder = io.opentelemetry.api.common.Attributes.builder();
+            attributes.getAttributesMap().forEach((x, y) -> addSpanAttribute(x, y, attributesBuilder));
+            return attributesBuilder.build();
+        } else {
+            return io.opentelemetry.api.common.Attributes.builder().build();
+        }
+    }
+
+    io.opentelemetry.api.trace.Span otelSpan(String spanName, Span parentOTelSpan, io.opentelemetry.api.common.Attributes attributes) {
         return parentOTelSpan == null || !(parentOTelSpan instanceof OTelSpan)
-            ? otelTracer.spanBuilder(spanName).startSpan()
-            : otelTracer.spanBuilder(spanName).setParent(Context.current().with(((OTelSpan) parentOTelSpan).getDelegateSpan())).startSpan();
+            ? otelTracer.spanBuilder(spanName).setAllAttributes(attributes).startSpan()
+            : otelTracer.spanBuilder(spanName)
+                .setParent(Context.current().with(((OTelSpan) parentOTelSpan).getDelegateSpan()))
+                .setAllAttributes(attributes)
+                .startSpan();
+    }
+
+    private void addSpanAttribute(AttributeKey key, Object value, AttributesBuilder attributesBuilder) {
+        switch (key.getType()) {
```

Comment:
```suggestion
        if (value instanceof Boolean) {
         ...
        } ....
```

Reply:
Please move it to Span::addAttributes

Reply:
Actually, this we need these attributes in the startSpan call, before even span creation. So adding to span wont serve the purpose here.

Reply:
I think you could exact that into package `OTelAttributesConverter` class or alike

Reply:
Yes, we can do that but it would still be duplication, since we would be having separate implementation for MockTelemetry. Let me see.